scipy/optimize/minpack.py

```diff
@@ -510,28 +523,34 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
         values will all be 1 (if the number of parameters for the function
         can be determined using introspection, otherwise a ValueError
         is raised).
-    sigma : None or M-length sequence, optional
-        If not None, the uncertainties in the ydata array. These are used as
-        weights in the least-squares problem
-        i.e. minimising ``np.sum( ((f(xdata, *popt) - ydata) / sigma)**2 )``
-        If None, the uncertainties are assumed to be 1.
+    sigma : None or M-length sequence or MxM array, optional
+        Determines the uncertainty in `ydata`.
+        If we define measurement errors as
+        ``e = ydata - f(xdata, *popt)``, then
+
+            - If `sigma` is 1-d, it determines standard deviations of errors `e`
+              and the optimized function is ``sum((e / sigma) ** 2)``.
+            - If `sigma` is 2-d, it determines the covariance matrix of errors `e`
+              and the optimized function is ``e.T.dot(inv(sigma)).dot(e)``.
+
+        None (default) is equivalent of 1-d `sigma` filled with ones.
     absolute_sigma : bool, optional
-        If False, `sigma` denotes relative weights of the data points.
-        The returned covariance matrix `pcov` is based on *estimated*
-        errors in the data, and is not affected by the overall
-        magnitude of the values in `sigma`. Only the relative
-        magnitudes of the `sigma` values matter.
-
-        If True, `sigma` describes one standard deviation errors of
-        the input data points. The estimated covariance in `pcov` is
-        based on these values.
+        If False, `sigma` is used in a relative sense. The returned parameter
```

Comment:
The description starts with "is based on scaled errors", but it is not clear what are these "scaled errors", assuming you read for the first time and didn't think it through. How are they scaled? Why they were scaled? 

Basically if we shorten the sequence: "The returned parameter covariance matrix `pcov` is based on *scaled* errors.", it just doesn't give the gist of what's going on in my opinion. And then it is sort of hard to perceive what follows. 

2. I meant that the optimization problem is solved in terms of scaled errors already (by covariance matrix or individual sd). And here another scaling is mentioned. Does it make sense? Or am I just misunderstanding something?

3. Yes, it is the same. Sample variance is just estimation of variance from a distribution sample https://en.wikipedia.org/wiki/Variance#Sample_variance We correct denominator by the number of parameters, but I don't consider it to be significant. I think of it as just a sample variance.

Still I don't claim that my view is correct.